File: Allowed.Maui.Hybridizer.Sample/ClientApp/src/plugins/ClientLifetimePlugin.ts
import {bridgeService} from "../hybridizer/BridgeService";
import {PlatformConstants} from "../hybridizer/PlatformConstants";
import {platformService} from "../hybridizer/PlatformService";

export class ClientLifetimePlugin {
    async initialize() {
        if (platformService.getPlatform() != PlatformConstants.Web)
            await bridgeService.invoke('ClientLifetime', 'Initialize');
    }

    async dispose() {
        if (platformService.getPlatform() != PlatformConstants.Web)
            await bridgeService.invoke('ClientLifetime', 'Dispose');
    }
}

export const clientLifetimePlugin = new ClientLifetimePlugin();

File: Allowed.Maui.Hybridizer.Essentials/Plugins/ClientLifetimePlugin.cs
using Allowed.Maui.Hybridizer.Abstractions.Plugins;

namespace Allowed.Maui.Hybridizer.Essentials.Plugins;

[HwvPlugin]
public class ClientLifetimePlugin(IServiceProvider serviceProvider, ClientLifetimeMethods methods)
{
    [HwvMethod]
    public void Initialize() => methods.OnInitialized?.Invoke(serviceProvider);

    [HwvMethod]
    public void Dispose() => methods.OnDisposed?.Invoke(serviceProvider);
}

public class ClientLifetimeMethods(
    Action<IServiceProvider>? onInitialized = null,
    Action<IServiceProvider>? onDisposed = null)
{
    public Action<IServiceProvider>? OnInitialized { get; set; } = onInitialized;
    public Action<IServiceProvider>? OnDisposed { get; set; } = onDisposed;
}